### PR TITLE
Build: add entry points without top level side-effects

### PR DESCRIPTION
**Pull request by OEvgeny**
> Fixes #

### Changed

- Excluded side effects from module entry points to prevent global scope modification, in PR [#5329](https://github.com/microsoft/BotFramework-WebChat/pull/5329), by [@OEvgeny](https://github.com/OEvgeny)

## Description

This PR improves the bundle architecture by migrating from Webpack to tsup and restructuring entry points to eliminate top-level side effects.

## Design

The solution implements a cleaner separation between bundle and module code:

1. `/module/exports*.ts` - Pure ES/CJS modules without side effects
   - Export Web Chat API surface
   - Maintain module-level encapsulation
   - Enable tree-shaking optimizations

2. `/bundle/index*.ts` - Bundle entry points with controlled side effects
   - Handle global `window.WebChat` registration
   - Manage version metadata
   - Import and expose module exports

This approach:
- Separates concerns between module API and bundle initialization
- Improves predictability of side effects
- Makes the codebase more maintainable
- Enables better tree-shaking in downstream applications

## Specific Changes

- Migrated build system from Webpack to tsup
- Restructured bundle package layout:
  - Added `src/module/exports.ts`, `exports-es5.ts`, `exports-minimal.ts`
  - Added `src/bundle/index.ts`, `index-es5.ts`, `index-minimal.ts`
  - Moved `polyfill.ts` to `src/bundle`
- Updated package.json:
  - Removed Webpack-related dependencies
  - Updated build scripts to use tsup
- Improved type definitions
- Added React resolution through `isomorphic-react`

## -

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [x] ~I have updated documentation~

## Review Checklist

-  [x] ~Accessibility reviewed (tab order, content readability, alt text, color contrast)~
-  [x] Browser and platform compatibilities reviewed
-  [x] ~CSS styles reviewed (minimal rules, no `z-index`)~
-  [x] ~Documents reviewed (docs, samples, live demo)~
-  [x] ~Internationalization reviewed (strings, unit formatting)~
-  [x] `package.json` and `package-lock.json` reviewed
-  [x] ~Security reviewed (no data URIs, check for nonce leak)~
-  [x] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/bundle/src/module/exports-minimal.ts [hunk 1]
```diff
@@ -13,40 +13,51 @@ import ReactWebChat, {
...
+const buildInfo = Object.freeze({
+  buildTool,
+  moduleFormat,
+  variant: 'minimal',
+  version
+});
 
-export const createDirectLine = options => {
-  options.botAgent &&
+export const createDirectLine = (options: Omit<Parameters<typeof defaultCreateDirectLine>[0], 'botAgent'>) => {
```
Comment (compulim) [inline packages/bundle/src/module/exports-minimal.ts:34]: ❤️

File: packages/component/src/providers/ActivityTree/ActivityTreeComposer.tsx [hunk 2]
```diff
@@ -25,22 +25,28 @@ const ActivityTreeComposer = ({ children }: ActivityTreeComposerProps) => {
   const [rawActivities] = useActivities();
   const getActivitiesByKey = useGetActivitiesByKey();
   const getKeyByActivity = useGetKeyByActivity();
+  const activityKeys = useActivityKeys();
 
   const activities = useMemo<readonly WebChatActivity[]>(() => {
     const activities: WebChatActivity[] = [];
 
+    if (!activityKeys) {
```
Comment (compulim) [inline packages/component/src/providers/ActivityTree/ActivityTreeComposer.tsx:33]: Shouldn't the call below `getKeyByActivity()` would return `undefined`?

**Review by compulim — APPROVED** [review]
I like that we don't need to export global in a way that we dupe export as modules.

File: packages/component/src/providers/ActivityTree/ActivityTreeComposer.tsx (hunk 2 above)
Comment (OEvgeny) [reply packages/component/src/providers/ActivityTree/ActivityTreeComposer.tsx:33]: We do this to ensure we update the memo when activityKeys got updated. I should've added a comment on this.

Comment (compulim) [reply packages/component/src/providers/ActivityTree/ActivityTreeComposer.tsx:33]: Oh wow, this is good. 👍🏻

**Review by compulim — APPROVED** [review]

**Pull request merged**